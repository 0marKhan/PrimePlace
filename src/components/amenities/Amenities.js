import React from "react";
import Chip from "@mui/material/Chip";

const Amenities = (props) => {
  return (
    <div>
      {props.amenities.map((item, index) => (
        <Chip key={index} className="mr-1 mb-2" label={item} />
      ))}
    </div>
  );
};

export default Amenities;
