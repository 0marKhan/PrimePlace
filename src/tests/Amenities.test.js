import React from "react";
import { render } from "@testing-library/react";
import Amenities from "../components/Amenities";

describe("Amenities Component", () => {
  it("renders a list of chips", () => {
    const amenities = ["Swimming Pool", "Gym", "Parking"];

    const { getByText } = render(<Amenities amenities={amenities} />);

    // Check if each amenity chip is rendered
    amenities.forEach((amenity) => {
      const chip = getByText(amenity);
      expect(chip).toBeInTheDocument();
    });
  });

  it("matches snapshot", () => {
    const amenities = ["Swimming Pool", "Gym", "Parking"];

    const { container } = render(<Amenities amenities={amenities} />);

    // Check if the rendered component matches the snapshot
    expect(container).toMatchSnapshot();
  });
});
