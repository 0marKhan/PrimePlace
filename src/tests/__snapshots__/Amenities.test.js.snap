// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Amenities Component matches snapshot 1`] = `
<div>
  <div>
    <div
      class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault mr-1 mb-2 css-1e8dhvr-MuiChip-root"
    >
      <span
        class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
      >
        Swimming Pool
      </span>
    </div>
    <div
      class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault mr-1 mb-2 css-1e8dhvr-MuiChip-root"
    >
      <span
        class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
      >
        Gym
      </span>
    </div>
    <div
      class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault mr-1 mb-2 css-1e8dhvr-MuiChip-root"
    >
      <span
        class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
      >
        Parking
      </span>
    </div>
  </div>
</div>
`;
